lib/grpc/client/adapters/mint/connection_process/connection_process.ex

```diff
@@ -261,6 +308,14 @@ defmodule GRPC.Client.Adapters.Mint.ConnectionProcess do
     new_state
   end
 
+  defp process_response({:error, request_ref, error}, state) do
+    pid = State.stream_response_pid(state, request_ref)
+    :ok = StreamResponseProcess.consume(pid, :error, error)
+    :ok = StreamResponseProcess.done(pid)
```

Comment:
Why you're sending `done/1` here? By default, even if a error happens, the connection will produce a `:done` message when the stream is closed by the server